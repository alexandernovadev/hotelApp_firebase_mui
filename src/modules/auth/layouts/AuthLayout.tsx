import { ReactJSXElement } from "@emotion/react/types/jsx-namespace";
import { Grid, Typography } from "@mui/material";
import { FlagTranslate } from "../../../UI/FlagTranslate";

interface Props{
  children: ReactJSXElement | ReactJSXElement[];
  title:string
}

export const AuthLayout = ({ children, title = "HILA" }:Props) => {
  return (
    <Grid
      container
      spacing={0}
      direction="column"
      alignItems="center"
      justifyContent="center"
      sx={{ 
        minHeight: "100vh",
        backgroundSize:"100% 100%",
        backgroundImage:"url('https://vaovatravel.com/wp-content/uploads/2022/06/YPO-Colombia-2022-419-1-1-scaled.jpg')",
        padding: 4 
      }}
    >
      <Grid
        item
        className="box-shadow animate__animated animate__fadeIn animate__faster"
        xs={3}
        sx={{
          width: { sm: 450 },
          backgroundColor: "primary.light",
          padding: 3,
          borderRadius: 2,
        }}
      >
        <FlagTranslate />
        <Typography variant="h5" sx={{ mb: 1 }}>
          {title}
        </Typography>
        {children}
      </Grid>
    </Grid>
  );
};
