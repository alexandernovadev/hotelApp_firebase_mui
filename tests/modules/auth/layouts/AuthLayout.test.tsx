import React from "react";
import { render, screen } from "@testing-library/react";
import { AuthLayout } from "../../../../src/modules/auth/layouts/AuthLayout";
jest.mock("react-i18next", () => ({
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

describe("Test in <AuthLayot/>", () => {
  const title = "Testing";
  const Componetn = () => <>Hola</>;

  test("Should match with the snapshot", () => {
    const { container } = render(
      <AuthLayout title={title} children={<Componetn />} />
    );
    expect(container).toMatchSnapshot();
  });

  test("Should show the correct title", () => {
    render(<AuthLayout title={title} children={<Componetn />} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
