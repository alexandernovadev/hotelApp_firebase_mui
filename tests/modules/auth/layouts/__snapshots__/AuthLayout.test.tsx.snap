// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <AuthLayot/> Should match with the snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-zqh5bb-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 box-shadow animate__animated animate__fadeIn animate__faster css-8ez7fo-MuiGrid-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1jcm1rz-MuiTypography-root"
      >
        Testing
      </h5>
      Hola
      <button
        class="css-crr5sf"
      >
        en
      </button>
      <b>
        |
      </b>
      <button
        class="css-crr5sf"
      >
        es
      </button>
    </div>
  </div>
</div>
`;
